ingestor-default/src/main/scala/za/co/absa/hyperdrive/ingestor/implementation/writer/kafka/KafkaStreamWriter.scala

```diff
@@ -32,11 +32,11 @@ private[writer] class KafkaStreamWriter(topic: String,
   def write(dataFrame: DataFrame): StreamingQuery = {
     dataFrame
       .writeStream
-      .options(extraOptions)
       .option(StreamWriterProperties.CheckpointLocation, checkpointLocation)
       .trigger(trigger)
       .option("topic", topic)
       .option("kafka.bootstrap.servers", brokers)
+      .options(extraOptions)
```

Comment:
I have to ask. Does order here (or in ParquetStreamWritter) really matter?

Reply:
Looks like it does matter.

From the DataFrameWriter doc(https://spark.apache.org/docs/latest/api/java/org/apache/spark/sql/DataFrameWriter.html):

> All options are maintained in a case-insensitive way in terms of key names. If a new option has the same key case-insensitively, it will override the existing option.

I tried a simple test overriding compression codec option when writing parquet file and results seem to confirm it.

Reply:
Thank you!